evalml/pipelines/components/transformers/encoders/ordinal_encoder.py

```diff
@@ -0,0 +1,322 @@
+    ]
+
+
+class OrdinalEncoder(Transformer, metaclass=OrdinalEncoderMeta):
+    """A transformer that encodes ordinal features as an array of ordinal integers representing the relative order of categories.
+
+    Args:
+        top_n (int): Number of categories per column to encode. If None, all categories will be encoded.
+            Otherwise, the `n` most frequent will be encoded and all others will be handled as unknown values.
+            To not have unknown values raise an error, set handle_unknown to "use_encoded_value".
+            Defaults to 10.
+        features_to_encode (list[str]): List of columns to encode. All other columns will remain untouched.
+            If None, all appropriate columns will be encoded. Defaults to None. The order of columns does not matter.
+        categories (list[list[str]]): A two dimensional list of categories, where `categories[i]` is a list of the categories
+            for the column at index `i` in the dataframes passed in at fit and transform.
+            The order of categories specified for a column does not matter.
+            Any category not present in categories will be handled as an unknown value.
+            To not have unknown values raise an error, set handle_unknown to "use_encoded_value".
+            This can also be `None` or `"auto"` if `top_n` is not None. Cannot be specified if top_n
+            is specified. Defaults to None.
+        handle_unknown ("error" or "use_encoded_value"): Whether to ignore or error for unknown categories
+            for a feature encountered during `fit` or `transform`. When set to "error",
+            an error will be raised when an unknown category is found.
+            When set to "use_encoded_value", unknown categories will be encoded as the value given
+            for the parameter unknown_value. Defaults to "error."
+        unknown_value (int or np.nan): The value to use for unknown categories seen during fit or transform.
+            Required when the parameter handle_unknown is set to "use_encoded_value."
+            The value has to be distinct from the values used to encode any of the categories in fit.
+            Defaults to None.
+        encoded_missing_value (int or np.nan): The value to use for missing (null) values seen during
+            fit or transform. Defaults to np.nan.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    name = "Ordinal Encoder"
+    hyperparameter_ranges = {}
+    """{}"""
+
+    def __init__(
+        self,
+        top_n=10,
+        features_to_encode=None,
+        categories=None,
+        handle_unknown="error",
+        unknown_value=None,
+        encoded_missing_value=None,
+        random_seed=0,
+        **kwargs,
+    ):
+        parameters = {
+            "top_n": top_n,
+            "features_to_encode": features_to_encode,
+            "categories": categories,
+            "handle_unknown": handle_unknown,
```

Comment:
I didn't include a `handle_missing` parameter, since "as_category" isn't an option here, but it means that we don't have the option to error when nans are seen. 

Do we want a `handle_missing` parameter that is either "use_encoded_value" or "error" and then pairs with the encoded_missing_value parameter?